test/e2e/device_test.go

```diff
@@ -0,0 +1,223 @@
+package e2e
+
+import (
+	"encoding/csv"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/devices"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCli_Devices(t *testing.T) {
+	var firstID string
+	subCommand := "device"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "create",
+			fields: fields{
+				MainCmd:  devices.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "create", "-p", os.Getenv("METAL_PROJECT_ID"), "-P", "c3.small.x86", "-m", "da", "-H", "metal-test-stag1", "-O", "ubuntu_20_04"})
```

Comment:
Instead of reading a `METAL_PROJECT_ID` from the environment, this test should first create a project with a standardized name (e.g., `metal-cli-test-device-<random suffix>`), and then create the test device(s) in that project.  The last device test should be a delete test so that the device is cleaned up as part of the test run to reduce the chance that our tests leave dangling resources.